compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/messages/CompilerMessageExtendedLocation.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.cli.common.messages
+
+import java.io.Serializable
+
+interface CompilerMessageExtendedLocation {
+    val path: String
+    val line: Int
+    val column: Int
+    val lineEnd: Int
+    val columnEnd: Int
+    val lineContent: String?
+
+    companion object {
+        @JvmStatic
+        fun create(path: String?): CompilerMessageExtendedLocation? =
+            create(path, -1, -1, null)
+
+        @JvmStatic
+        fun create(path: String?, line: Int, column: Int, lineContent: String?): CompilerMessageExtendedLocation? =
+            if (path == null) null else CompilerMessageExtendedLocationImpl(path, line, column, -1, -1, lineContent)
+
+        @JvmStatic
+        fun create(
+            path: String?,
+            lineStart: Int,
+            columnStart: Int,
+            lineEnd: Int?,
+            columnEnd: Int?,
+            lineContent: String?
+        ): CompilerMessageExtendedLocation? =
+            if (path == null) null else CompilerMessageExtendedLocationImpl(
+                path,
+                lineStart,
+                columnStart,
+                lineEnd ?: -1,
+                columnEnd ?: -1,
+                lineContent
+            )
+    }
+}
+
+private data class CompilerMessageExtendedLocationImpl(
+    override val path: String,
+    override val line: Int,
+    override val column: Int,
+    override val lineEnd: Int,
+    override val columnEnd: Int,
+    override val lineContent: String?,
+) : Serializable, CompilerMessageExtendedLocation {
+    override fun toString(): String {
+        val start =
+            if (line == -1 && column == -1) ""
+            else "$line:$column"
+        val end =
+            if (lineEnd == -1 && columnEnd == -1) ""
+            else if (lineEnd == line) " - $columnEnd"
+            else " - $lineEnd:$columnEnd"
+        val loc = if (start.isEmpty() && end.isEmpty()) "" else " ($start$end)"
+        return path + loc
+    }
+
+    companion object {
+        private const val serialVersionUID: Long = 8228357580L
```

Comment:
With the new serializables, we start now with `serialVersionUID = 0L` and increment. The pseudo-random big numbers like these were used a long time ago and now considered unnecessary.